logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -55,29 +58,42 @@ internal class ConnectionDataSource(
         return wrapApiRequest {
             connectionApi.createConnection(idMapper.toApiModel(userId))
         }.map { connection ->
-            val connectionSent = connection.copy(status = ConnectionState.SENT)
+            val connectionSent = connection.copy(status = ConnectionStateDTO.SENT)
             updateUserConnectionStatus(listOf(connectionSent))
         }
     }
 
-    private fun connectionStateToDao(state: ConnectionState): UserEntity.ConnectionState = when (state) {
-        ConnectionState.PENDING -> UserEntity.ConnectionState.PENDING
-        ConnectionState.SENT -> UserEntity.ConnectionState.SENT
-        ConnectionState.BLOCKED -> UserEntity.ConnectionState.BLOCKED
-        ConnectionState.IGNORED -> UserEntity.ConnectionState.IGNORED
-        ConnectionState.CANCELLED -> UserEntity.ConnectionState.CANCELLED
-        ConnectionState.MISSING_LEGALHOLD_CONSENT -> UserEntity.ConnectionState.MISSING_LEGALHOLD_CONSENT
-        ConnectionState.ACCEPTED -> UserEntity.ConnectionState.ACCEPTED
+    override suspend fun updateConnectionStatus(userId: UserId, connectionState: ConnectionState): Either<CoreFailure, Unit> {
+        // Check if we can transition to the correct connection status
+        val canTransitionToStatus = checkIfCanTransitionToConnectionStatus(connectionState)
+        val newConnectionStatus = connectionStatusMapper.connectionStateToApi(connectionState)
+        if (!canTransitionToStatus || newConnectionStatus == null) {
+            return Either.Left(InvalidMappingFailure)
+        }
+
+        return wrapApiRequest {
+            connectionApi.updateConnection(idMapper.toApiModel(userId), newConnectionStatus)
+        }.map { connection ->
+            val connectionSent = connection.copy(status = newConnectionStatus)
+            updateUserConnectionStatus(listOf(connectionSent))
+        }
+    }
+
+    private fun checkIfCanTransitionToConnectionStatus(connectionState: ConnectionState): Boolean = when (connectionState) {
+        ConnectionState.IGNORED -> false // TODO: implement and move to next case
+        ConnectionState.CANCELLED, ConnectionState.ACCEPTED -> true
+        else -> false
```

Comment:
I'm not sure I understand this method. Did you mean to call it `checkIfCanTransitionToConnectedStatus` ? Otherwise, if it is just checking if the transition between connection statuses makes sense, wouldn't we also need to provide the current connection status besides the new one?